src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,489 @@
+            /// <param name="value">The read-only span containing the characters to be written.</param>
+            private async ValueTask WritingCharValuesInChunksWithoutEscapingAsync(ReadOnlyMemory<char> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += ODataUtf8JsonWriter.chunkSize)
+                {
+                    int remainingChars = Math.Min(ODataUtf8JsonWriter.chunkSize, value.Length - i);
+                    bool isFinalBlock = false;
+                    ReadOnlyMemory<char> chunk = value.Slice(i, remainingChars);
+
+                    // If the buffer is not empty, then we copy the chars from the buffer
+                    // to the current chunk being processed.
+                    if (numOfCharsNotWrittenFromPreviousChunk > 0)
+                    {
+                        // Get unprocessed chars from the buffer.
+                        ReadOnlyMemory<char> charsNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+                        int totalLength = charsNotProcessedFromPreviousChunk.Length + chunk.Length;
+
+                        char[] combinedArray = ArrayPool<char>.Shared.Rent(totalLength);
+
+                        // Copy chars from charsNotProcessedFromPreviousChunk to the combined array
+                        charsNotProcessedFromPreviousChunk.CopyTo(combinedArray);
+
+                        // Copy chars from chunk to the combined array, starting from the end of charsNotProcessedFromPreviousChunk
+                        chunk.CopyTo(combinedArray.AsMemory().Slice(charsNotProcessedFromPreviousChunk.Length));
+
+                        // Write the chunk.
+                        this.WriteChunkWithoutEscaping(combinedArray, totalLength, isFinalBlock);
+
+                        ArrayPool<char>.Shared.Return(combinedArray);
+                    }
+                    else
+                    {
+                        this.WriteChunkWithoutEscaping(chunk.Span, chunk.Length, isFinalBlock);
+                    }
+
+                    // Flush the writer if the buffer threshold is reached
+                    await this.jsonWriter.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+                }
+            }
+
+            private void WriteChunk(ReadOnlySpan<char> chunk, int chunkLength, int firstIndexToEscape, bool isFinalBlock)
+            {
+                if (firstIndexToEscape != -1)
+                {
+                    this.jsonWriter.WriteEscapedStringChunk(chunk, firstIndexToEscape, isFinalBlock, out numOfCharsNotWrittenFromPreviousChunk);
+
+                    if (this.numOfCharsNotWrittenFromPreviousChunk > 0)
+                    {
+                        if (this.buffer == null)
+                        {
+                            this.buffer = ArrayPool<char>.Shared.Rent(ODataUtf8JsonWriter.chunkSize);

```

Comment:
Why not:
```csharp
this.buffer = ArrayPool<byte>.Shared.Rent(this.numBytesNotWrittenFromPreviousChunk);
```

Reply:
https://github.com/OData/odata.net/pull/2880#discussion_r1551202855